src/box/lua/error.cc

```diff
@@ -228,6 +265,14 @@ luaT_error_create(lua_State *L, int top_base)
 		/* Remove the table. */
 		lua_pop(L, 1);
 	}
+	if (record != NULL && record->errfields_count > 0) {
+		if (top - top_base - 1 >= record->errfields_count)
+			luaL_error(L, "too many error arguments");
+		int erridx = 0;
+		for (int i = top_base + 1; i <= top; i++, erridx++)
+			luaT_error_client_encode_arg(L, error, i, erridx,
+						     record);
+	}
```

Comment:
I don't think we need a Lua API for setting the system error codes with payload fields or do we? In Lua we could set payload explicitly, I guess, because there aren't that many places where we need to do that AFAIK, but that's not for sure.

Reply:
Make Lua API on par with C in this respect as decided after discussion f2f.